vital/types/track_set.cxx

```diff
@@ -578,21 +580,18 @@ ::set_frame_data( track_set_frame_data_sptr data,
 
 track_set_implementation_uptr
 simple_track_set_implementation
-::clone() const
+::clone( clone_type ct ) const
 {
   std::unique_ptr<simple_track_set_implementation> new_stsi =
     std::unique_ptr<simple_track_set_implementation>(new simple_track_set_implementation());
 
-  for (auto trk : data_)
+  for ( auto const& trk : data_ )
   {
-    new_stsi->data_.push_back(trk->clone());
+    new_stsi->data_.emplace_back( trk->clone( ct ) );
   }
-  for (auto fd : frame_data_)
+  for ( auto const& fd : frame_data_ )
   {
-    if ( fd.second )
```

Comment:
same thing here.  I think the frame data can be null.

Reply:
Same as above, mostly. The difference here is that the initial map might be passed in by the user. Again, however, `data_` is in the same boat.

If you feel strongly about it, we can check both instead of neither... but checking one and not the other feels schizophrenic.